web/src/custom_profile_fields_ui.ts

```diff
@@ -246,18 +247,20 @@ export function initialize_custom_date_type_fields(element_id: string, user_id:
             );
             const original_value = people.get_custom_profile_data(user_id, field_id)?.value ?? "";
             instance.setDate(original_value);
-            if (user_id !== people.my_current_user_id()) {
-                // Trigger "input" event so that save button state can
-                // be toggled in "Manage user" modal.
-                $input_elem
-                    .closest(".custom_user_field")
-                    .find(".date-field-alt-input")
-                    .trigger("input");
-            }
+            // Trigger "input" event so that save button state can
+            // be toggled in "Manage user" modal.
+            $input_elem
+                .closest(".custom_user_field")
+                .find(".date-field-alt-input")
+                .trigger("input");
```

Comment:
This will now also be run for "Profile" panel when it isn't needed.

Also, a better way to handle the "Profile" panel cases separately would be to just pass a `for_profile_settings_panel` boolean to `initialize_custom_date_type_fields` and use that instead of using if block for user_id and element_id.